number/src/traits.rs

```diff
@@ -0,0 +1,33 @@
+use num_traits::Zero;
+use std::{
+    fmt,
+    ops::{Add, AddAssign, Div, Mul, Neg, Sub},
+};
+
+use crate::{AbstractNumberType, DegreeType};
+pub trait FieldElementTrait:
```

Comment:
Why do we need a new trait instead of just using something from ark?

Reply:
We currently already wrap ark. This PR just moves the methods to a trait. If we wanna use ark directly we need to change the call sites of these methods to using ark only. Also (less important) this lets us support something else than ark if needed by just implementing the trait.

Reply:
ok